src/Helper.js

```diff
@@ -0,0 +1,311 @@
+      return device.area_id === area.area_id;
+    }).map(device => {
+      return device.id;
+    });
+
+    // Return the entities of which all conditions below are met:
+    // 1. The entity is linked to a device which is linked to the given area,
+    //    or the entity itself is linked to the given area.
+    // 2. The entity's domain matches the given domain.
+    // 3. The entity is not hidden and is not disabled.
+    return this.#entities.filter(entity => {
+      return (
+          (areaDeviceIds.includes(entity.device_id) || entity.area_id === area.area_id)
+          && entity.entity_id.startsWith(`${domain}.`)
+          && entity.hidden_by == null && entity.disabled_by == null
+      );
+    });
+  }
+
+  /**
+   * Get state entities, filtered by area and domain.
+   *
+   * The result excludes hidden and disabled entities.
+   *
+   * @param {areaEntity} area Area entity.
+   * @param {string} domain Domain of the entity-id.
+   *
+   * @return {stateObject[]} Array of state entities.
+   */
+  static getStateEntities(area, domain) {
+    const states = [];
+
+    // Create a map for the hassEntities and devices {id: object} to improve lookup speed.
+    /** @type {Object<string, hassEntity>} */
+    const entityMap = Object.fromEntries(this.#entities.map(entity => [entity.entity_id, entity]));
+    /** @type {Object<string, deviceEntity>} */
+    const deviceMap = Object.fromEntries(this.#devices.map(device => [device.id, device]));
+
+    // Get states whose entity-id starts with the given string.
+    const stateEntities = Object.values(this.#hassStates).filter(
+        state => state.entity_id.startsWith(`${domain}.`),
+    );
+
+    for (const state of stateEntities) {
+      const hassEntity = entityMap[state.entity_id];
+      const device     = deviceMap[hassEntity?.device_id];
+
+      // TODO: Agree on conditions (https://github.com/AalianKhan/mushroom-strategy/pull/7#discussion_r1173032335)
+      // Collect states of which any (whichever comes first) of the conditions below are met:
+      // 1. The linked entity is linked to the given area.
+      // 2. The entity is linked to a device, and the linked device is linked to the given area.
+      if (
+          (hassEntity?.area_id === area.area_id)
+          || (device && device.area_id === area.area_id)
+      ) {
+        states.push(state);
+      }
+
+      /*
+       // Collect states of which all conditions below are met:
+       // 1. The linked entity is linked to the given area or isn't linked to any area.
+       // 2. The linked device (if any) is assigned to the given area.
+       if (
+       (!hassEntity?.area_id || hassEntity.area_id === area.area_id)
+       && (device && device.area_id === area.area_id)
+       ) {
+       states.push(state);
+       }
+       */
+    }
+
+    return states;
+  }
+
+  /**
+   * Sanitize a classname.
+   *
+   * The name is sanitized nu upper-casing the first character of the name or after an underscore.
```

Comment:
What does this `nu` stand for? 
I don't understand the purpose of this function and where it will be used. Can you clarify?

Reply:
No idea why it says "nu", it should by "by".

It's used in several other files to convert a string into a format suitable to import different classes (files).
By convention the filename of classes are Upper Camel cased.
E.g. `foo_bar` to `FooBar`.